opac/webapp/main/views.py

```diff
@@ -1424,10 +1424,18 @@ def article_detail_pdf(url_seg, url_seg_issue, url_seg_article, lang_code=''):
 @cache.cached(key_prefix=cache_key_with_lang_with_qs)
 def router_legacy_pdf(journal_acron, issue_info, pdf_filename):
     pdf_filename = '%s.pdf' % pdf_filename
+
+    journal = controllers.get_journal_by_url_seg(journal_acron)
+
+    if not journal:
+        abort(404, _('Periódico não encontrado'))
```

Comment:
@gitnnolabs sugiro erro 400 Bad request (https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors)
E a mensagem deve ser algo como "Este PDF não é de um periódico da coleção www.scielo.br. Consulte search.scielo.org"

Reply:
@robertatakenaka pensei nisso, porém o cache não "estoca" **Bad request 400**, por isso a opção em **404** que é cacheado, dessa a requisição não vai até o backend.

Até pensei em uma mensagem mais elaborada, e indicar outro site do SciELO, mas não é garantido que estejamos realmente com o periódico ausente e também não é garantido que tenha qualquer coisa na (URL) que não seja um acrônimo e não sei ficamos indicando olhar na rede SciELO se enquadra em todas as situações que essa, URL pode ter.  

Enfim se achar melhor, realizamos a alteração da msg!, mas do `status code` não aliviaria o backend, já o 404 é cacheado.

Reply:
OK para a primeira parte. Mas melhor sim uma mensagem mais elaborada pois fica parecendo falha no próprio site e 100% dos casos eram de periódicos fora de scielo.br. A estratégia de indicar o search.scielo.org é porque muitos usuários podem chegar via google ou outra base de dados que até desconhece scielo. E chegar e não encontrar sem nenhuma outra informação é muito ruim, pelo menos obter o link para o search já faz o usuário retornar, tentar de novo.

Reply:
Outro ponto importante é que ao retornarmos **Bad request 400**, estamos indicando algo de errado com a requisição ou a formação da (URL) o que não é verdade, está tudo correto, somente não foi encontrado o dado para o recurso acessado. 

Link para 400 no msdn: https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/400

Reply:
@robertatakenaka o que acha da msg: 

> Não encontramos esse PDF nessa coleção. Sugerimos que consulte nossa rede, acessando: https://www.scielo.org

Reply:
@gitnnolabs do meu ponto de vista é sim uma bad request já que a sintaxe está errada, o acronimo é inexistente.

Reply:
@gitnnolabs já tinha sugerido: "Este PDF não é de um periódico da coleção www.scielo.br. Consulte search.scielo.org"
Outra opção:

- "Este PDF não existe em www.scielo.br. Consulte search.scielo.org"

Nota: É melhor em "não existe" no lugar "não encontrado", pois este pode parecer que um dia será encontrado.

Reply:
@gitnnolabs não usaria "coleção", pois talvez para o usuário não fica claro o que é coleção, mas pode trocar coleção por "site"

Reply:
@robertatakenaka estou aguardando a infra me responder se o 400 é cacheado!

Reply:
Segue link para um artigo que afirma que o 404 é cacheado: https://www.section.io/blog/varnish-cache-404-errors/

Reply:
Status cacheados: 

- 200: OK
- 203: Non-Authoritative Information
- 300: Multiple Choices
- 301: Moved Permanently
- 302: Moved Temporarily
- 304: Not modified
- 307: Temporary Redirect 
- 410: Gone
- 404: Not Found